src/index.js

```diff
@@ -150,102 +151,123 @@ export default function graphqlHTTP(options: Options): Middleware {
       // Parse the Request body.
       return parseBody(request);
     }).then(bodyData => {
-      const urlData = request.url && url.parse(request.url, true).query || {};
-      showGraphiQL = graphiql && canDisplayGraphiQL(request, urlData, bodyData);
-
-      // Get GraphQL params from the request and POST body data.
-      const params = getGraphQLParams(urlData, bodyData);
-      query = params.query;
-      variables = params.variables;
-      operationName = params.operationName;
-
-      // If there is no query, but GraphiQL will be displayed, do not produce
-      // a result, otherwise return a 400: Bad Request.
-      if (!query) {
-        if (showGraphiQL) {
-          return null;
+      function executeQuery(requestData) {
```

Comment:
Well, express-graphql already creates a bunch of functions for every request to handle promise continuations. Ideally these all go away in a young gen GC anyway.